File: delphi/witness_seed_3.0/src/Kairos.pas
unit Kairos;

interface

procedure UpdateCoherence(var I: array of Double; const Phase: Double);

implementation

procedure UpdateCoherence(var I: array of Double; const Phase: Double);
var
  I: Integer;
begin
  for I := Low(I) to High(I) do
    I[I] := I[I] * Cos(Phase);
end;

end.

File: fortran/witness_seed_3.0/src/kairos.f90
module kairos
  use iso_fortran_env, only: real64
  implicit none
contains
  subroutine update_coherence(I, phase)
    real(real64), intent(inout) :: I(:)
    real(real64), intent(in) :: phase
    I = I * cos(phase)
  end subroutine
end module kairos